contributed_definitions/nyaml/NXopt.yaml

```diff
@@ -427,152 +750,119 @@ NXopt(NXobject):
-          and provided in 'measured_data'.
-        dimensions:
-          rank: 3
-          dim: [[1, N_measurements], [2, N_observables], [3, N_spectrum]]
-        \@units:
+          Arbirary sample property which may be varied during the experiment
+          and controlled by a device. Examples are pressure, voltage, magnetic field etc.
+          Similar to the temperautre description of the sample.
+        (NXsensor):
           exists: optional
-          doc: |
-            If applicable, change 'unit: NX_ANY' to the appropriate NXDL unit.
-            If the unit of the measured data is not covered by NXDL units state
-            here which unit was used.
-      varied_parameter_link:
+        (NXactuator):
+          exists: optional
+      thickness(NX_NUMBER):
         exists: optional
         doc: |
-          List of links to the values of the sensors. Add a link for each
-          varied parameter (i.e. for each sensor).
-        dimensions:
-          rank: 1
-          dim: [[1, N_sensors]]
-      reference_data_link:
+          (Measured) sample thickness.
+          
+          The information is recorded to qualify if the light used was likely
+          able to shine through the sample. 
+          
+          In this case the value should be set to the actual thickness of
+          the specimen viewed for an illumination situation where the nominal
+          surface normal of the specimen is parallel to the optical axis.
+        unit: NX_LENGTH
+      thickness_determination:
         exists: optional
         doc: |
-          Link to the NeXus file which describes the reference data if a
-          reference measurement was performed. Ideally, the reference
-          measurement was performed using the same conditions as the actual
-          measurement and should be as close in time to the actual measurement
-          as possible.
-      data_software(NXprocess):
+          If a thickness if given, please specify how this thickness was estimated or determined.
+      layer_structure:
         exists: optional
-        program:
-          doc: |
-            Commercial or otherwise defined given name of the program that was
-            used to generate the result file(s) with measured data and/or
-            metadata (in most cases, this is the same as INSTRUMENT/software).
-            If home written, one can provide the actual steps in the NOTE
-            subfield here.
-        version:
-          doc: |
-            Either version with build number, commit hash, or description of a
-            (online) repository where the source code of the program and build
-            instructions can be found so that the program can be configured in
-            such a way that result files can be created ideally in a
-            deterministic manner.
-        \@url:
-          exists: optional
-          doc: |
-            Website of the software.
-      (NXdata):
+        doc: |
```

Comment:
Consider using `NXsample_component_set`.

Reply:
all right, just took an note for now.